include/domain_bridge/domain_bridge.hpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021, Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DOMAIN_BRIDGE__DOMAIN_BRIDGE_HPP_
+#define DOMAIN_BRIDGE__DOMAIN_BRIDGE_HPP_
+
+#include <memory>
+#include <string>
+
+#include "domain_bridge/topic_bridge_options.hpp"
+#include "domain_bridge/visibility_control.hpp"
+
+namespace rclcpp
+{
+class Executor;
+}
+
+namespace domain_bridge
+{
+
+class DomainBridgeImpl;
+
+/// Bridge ROS entities across different domains.
+class DomainBridge
+{
+public:
+  /// Constructor.
+  DOMAIN_BRIDGE_PUBLIC
+  DomainBridge();
+
+  /// Destructor.
+  DOMAIN_BRIDGE_PUBLIC
+  ~DomainBridge();
+
+  /// Add the bridge to an executor.
+  /**
+   * When the executor is running ROS traffic will be bridged between domains.
+   * Only topics registered with \ref bridge_topic() will have their messages forwarded.
+   *
+   * This method will add all nodes associated with this bridge to the provided executor.
+   *
+   * \param executor: The executor to add this domain bridge to.
+   */
+  DOMAIN_BRIDGE_PUBLIC
+  void add_to_executor(std::shared_ptr<rclcpp::Executor> executor);
```

Comment:
```suggestion
  void add_to_executor(rclcpp::Executor & executor);
```

no need of a shared ptr here

Reply:
a559080